.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,61 @@
+name: Docker Build and Push
+
+on:
+  schedule:
+    - cron: '13 13 * * *'
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Get latest commit ID from nginx/nginx
+        run: |
+          LATEST_COMMIT=$(curl -s https://api.github.com/repos/nginx/nginx/commits/master | jq -r '.sha')
+          echo "commit_id=$LATEST_COMMIT" >> $GITHUB_ENV
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Pull the latest image for comparison
+        run: |
+          docker pull ghcr.io/nginx/nginx-quic-qns:latest || true
```

Comment:
That sounds wasteful - maybe we can somehow pull the information we need without downloading the image?

Reply:
Thanks for the tip, @thresheek ! Yes, using `skopeo` is indeed more effective in this context.